src/vrapp/VRVolumeApp.cpp

```diff
@@ -750,15 +751,15 @@ void VRVolumeApp::animated_render(int tfn, int vol)
   bool use_tranferFunction = m_ui_view->is_use_transfer_function_enabled();
 
   size_t max_animation_length = 0;
-  float global_min = std::numeric_limits<float>::max();
-  float global_max = std::numeric_limits<float>::min();
+  float volume_min = std::numeric_limits<float>::max();
+  float volume_max = std::numeric_limits<float>::min();
   bool dirty = false;
   if (m_ui_view && m_ui_view->is_transfer_function_enabled(tfn, vol))
   {
     dirty = true;
     max_animation_length = std::max(max_animation_length, m_volumes[vol].size());
-    global_min = std::min(global_min, m_volumes[vol][active_volume]->getMin());
-    global_max = std::max(global_max, m_volumes[vol][active_volume]->getMax());
+    volume_min = std::min(volume_min, m_volumes[vol][active_volume]->getMin());
+    volume_max = std::max(volume_max, m_volumes[vol][active_volume]->getMax());
 
     if (active_volume < m_volumes[vol].size() && active_volume2 < m_volumes[vol].size() && m_volumes[vol][active_volume]->texture_initialized() && m_volumes[vol][active_volume2]->texture_initialized())
```

Comment:
Can you sparkle some comments on the logic? It a bit hard to follow what's going on. What's the meaning of `dirty`? Also why is this hardcoded to 2 volumes? Will this `if` statement work if there are 3 volumes loaded?

Reply:
I renamed variables to make them more meaningful to the logic.
I added comments on the code indicating what some of the variables do and how they workout the animation logic.

Reply:
Can you answer this question:

Also why is this hardcoded to 2 volumes? Will this if statement work if there are 3 volumes loaded?